sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSql.java

```diff
@@ -47,22 +36,23 @@
 Pipeline p = Pipeline.create(options);
 
 //create table from TextIO;
-PCollection<BeamSqlRow> inputTableA = p.apply(TextIO.read().from("/my/input/patha"))
-    .apply(...);
-PCollection<BeamSqlRow> inputTableB = p.apply(TextIO.read().from("/my/input/pathb"))
-    .apply(...);
+PCollection<Row> inputTableA = p.apply(TextIO.read().from("/my/input/patha")).apply(...);
+PCollection<Row> inputTableB = p.apply(TextIO.read().from("/my/input/pathb")).apply(...);
 
 //run a simple query, and register the output as a table in BeamSql;
 String sql1 = "select MY_FUNC(c1), c2 from PCOLLECTION";
-PCollection<BeamSqlRow> outputTableA = inputTableA.apply(
-    BeamSql.query(sql1)
-    .withUdf("MY_FUNC", MY_FUNC.class, "FUNC"));
+PCollection<Row> outputTableA = inputTableA.apply(
+    BeamSql
+        .query(sql1)
+        .registerUdf("MY_FUNC", MY_FUNC.class, "FUNC")
+        .toPTransform());
 
 //run a JOIN with one table from TextIO, and one table from another query
-PCollection<BeamSqlRow> outputTableB = PCollectionTuple.of(
-    new TupleTag<BeamSqlRow>("TABLE_O_A"), outputTableA)
-                .and(new TupleTag<BeamSqlRow>("TABLE_B"), inputTableB)
-    .apply(BeamSql.queryMulti("select * from TABLE_O_A JOIN TABLE_B where ..."));
+PCollection<Row> outputTableB =
+    PCollectionTuple
+        .of(new TupleTag<>("TABLE_O_A"), outputTableA)
```

Comment:
In this case it seems that the tuple tags aren't useful and could be simply strings. Not sure if the way to fix that is to allow `PCollectionTuple` to take a `String` instead of a `TupleTag`, or to introduce a new similar API in BeamSql for building `PCollectionTuple`s.

Reply:
I don't know what's the right way either. `PCollectionTuple` seems to be low level, and I am not sure it makes sense to expose overloads for `String` tags. Will try `String` overloads in this PR to see if it works.

Not sure whether it's better to expose an API in SQL for building `PCollectionTuple`s, its too generic. It will probably make more sense to implement a similar class specific to SQL use case. But that can come in conflict with schema-aware `PCollection`s work.

Reply:
Hm I don't think it would conflict with schema'd PCollections. On the other hand this is not a usability regression from before this PR so fine to keep as is for now, but I think it could look better as e.g. `BeamSql.table("foo", foo).and("bar", bar).apply(BeamSql.query("select * from foo join bar"))`

Reply:
FWIW, and I think we should add this to the guide, we should not make fluent APIs that try to mimic English grammar. Even for native English speakers (like me) they mask the underlying semantic objects in favor of some idea that the utterance should be interpreted as a whole, which is almost always a bad thing in programming.

Instead, fluent APIs are all about having one immutable object and deriving another from it via changing one little bit. So `.table` and `.and` in your example should both be the same name, something like `registerTable` or `addTable`.

Reply:
The point of `PCollectionTuple` is to allow a heterogeneous map where when you pull the `PCollection` back out, you have strong typing guarantees. (TBH I don't think that is so important anymore as pipeline validation time is when a lot of errors get caught)

The other benefit that `PCollectionTuple` can offer is that the tuples can have captured type information by using the `new TupleTag<XYZ>() {}` trick. We use this to infer coders, for example. (If Java reflection "just worked" then we would not need this). It isn't relevant for SQL on `PCollection<Row>`. But this connects with a conversation we had the other day: if you have `PCollection<MyPojo>` where `MyPojo` has a registered schema, then the SQL transform could accept it directly and insert necessary conversions to support table-oriented operations.